src/System.Private.ServiceModel/tests/Scenarios/Client/ClientBase/ClientBaseTests.cs

```diff
@@ -219,6 +219,93 @@ MessageHeader header
         }
     }
 
+    [Fact]
+    [OuterLoop]
+    public static void DefaultSettings_Echo_Cookie()
+    {
+        // *** SETUP *** \\
+        var factory = new ChannelFactory<IWcfAspNetCompatibleService>(
+                 new BasicHttpBinding()
+                 {
+                     AllowCookies = true
+                 },
+                 new EndpointAddress(Endpoints.HttpBaseAddress_Basic));
+
+        IWcfAspNetCompatibleService serviceProxy = factory.CreateChannel();
+
+        IHttpCookieContainerManager cookieManager = ((IChannel)serviceProxy).GetProperty<IHttpCookieContainerManager>();
+        Assert.True(cookieManager != null);
+        Assert.True(cookieManager.CookieContainer != null);
+        string cookieName = "cookieName";
+        string cookieValue = "cookieValue";
+        string cookieSentOut = $"{cookieName}={cookieValue}";
+        cookieManager.CookieContainer.Add(new Uri(Endpoints.HttpBaseAddress_Basic), new System.Net.Cookie(cookieName, cookieValue));
+
+        try
+        {
+            // *** EXECUTE *** \\
+            string cookieSentBack = serviceProxy.EchoCookie();
+
+            // *** VALIDATE *** \\
+            Assert.True(cookieSentOut == cookieSentBack,
+                $"The expected cookie sent back from the server was {cookieSentOut}, but the actuall cookie was {cookieSentBack}");
+
+            // *** CLEANUP *** \\
+            factory.Close();
+            ((ICommunicationObject)serviceProxy).Close();
+        }
+        finally
+        {
+            // *** ENSURE CLEANUP *** \\
+            ScenarioTestHelpers.CloseCommunicationObjects((ICommunicationObject)serviceProxy, factory);
+        }
+    }
+
+    [Fact]
+    [OuterLoop]
+    public static void DefaultSettings_SetCookieOnServerSide()
+    {
+        // *** SETUP *** \\
+        Uri uri = new Uri(Endpoints.HttpBaseAddress_Basic);
+        var factory = new ChannelFactory<IWcfAspNetCompatibleService>(
+                 new BasicHttpBinding()
+                 {
+                     AllowCookies = true
+                 },
+                 new EndpointAddress(uri));
+
+        IWcfAspNetCompatibleService serviceProxy = factory.CreateChannel();
+
+        IHttpCookieContainerManager cookieManager = ((IChannel)serviceProxy).GetProperty<IHttpCookieContainerManager>();
+        Assert.True(cookieManager != null);
+        Assert.True(cookieManager.CookieContainer != null);
+
```

Comment:
Whenever there is more than a single assert like this, you need a message to describe which one was triggered.  In a lab run it would not be obvious which failed.